### PR TITLE
Remove special handling of nonce method

### LINKED ISSUE #1005: eth_getTransactionCount returns 0x1 for newly deployed contract in previous block
Steps for reproducing:

 - `npx hardhat node`
 - `curl --data '{"id":6185245701164,"jsonrpc":"2.0","params":[{"from":"0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266","gas":"0x52080","gasPrice":"0x3B9ACA00","value":"0x1","data":"0x363d3d373d3d3d363d73bebebebebebebebebebebebebebebebebebebebe5af43d82803e903d91602b57fd5bf3","nonce": "0x0"}],"method":"eth_sendTransaction"}' -H "Content-Type: application/json"`
 - `curl --data '{"method":"eth_getTransactionCount","params":[{deployed_contract}, "0x0"],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json"`

The third step should return 0x0 but instead, the result is 0x1.

### PR DESCRIPTION
**Pull request by fvictorio**
Closes #1005.

### CODE REVIEW

**Comment by fvictorio** [comment]
The error that is failing here is fixed if I merge #1016 in this branch. So let's wait until that one is merged before reviewing this one.

Alternatively, I can make a single PR with both changes.

**Review by alcuadrado — APPROVED** [review]

**Pull request merged**
